### ISSUE TITLE
List of required APIs to ZPA

### ISSUE THREAD

**Issue by bennyboer** [Fachschaft07/Guide7#68]
Create a list of needed APIs to the ZPA!

**Comment by bennyboer** [comment]
Currently we are parsing the HTML of the notice board so it would be nice to have a better API for that. Additionally for the appointments we have no option but parsing HTMl of the appointment page of cs.hm.edu. Currently we use the iCal file of hm.edu for appointments which does not contain all relevant appointments for department 07.
In the future it would be cool to show the personal schedule for students as an addition to the week plan, although there is currently no API (as far as I know).

Summing up:
- API for notice board (Schwarzes Brett) (ZPA)
- API to fetch department 07 appointments (cs.hm.edu)
- API to get personal schedule (ZPA)

**Comment by thomai** [comment]
If you want (new) API endpoints, just send me a list via email with rough descriptions. I will talk to the responsible people of the department. They are quite cooperative. Please write in German.

**Comment by thomai** [comment]
The goal for this issue changed: We need a wishlist for the faculty employees regarding their efforts. They will provide us an API and they want to support us with the app in general (not only API interfaces).

**Comment by bennyboer** [comment]
Adding to the list:
- Lost and found (if that service does exist anymore)
- Multi-language support (german, english) for hm.edu services and ZPA (if it exists), since the app should support both (Right now the contents are only loaded in german)

@thomai I'm trying to complete the list of needed services by beginning of the next semester (18th of March). I want to wait for the FSS first to see if people have more ideas for features 😃. Otherwise we will have to send the current state with some added details.

**Comment by thomai** [comment]
@bennyboer I'd add a ZPA test user for developers but the credentials should be kept secret. Only core developers that are FS members may have access to it.

**Comment by bennyboer** [comment]
@thomai Sounds good, let's discuss that today during the FSS

**Comment by bennyboer** [comment]
Moving this issue to the March project because we wait for more feature ideas.

**Comment by thomai** [comment]
Please keep in mind that our main contact person of the faculty will be on vacation in a few months.
